packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -29,6 +29,15 @@ export 'semantics_event.dart' show SemanticsEvent;
 /// Used by [SemanticsNode.visitChildren].
 typedef SemanticsNodeVisitor = bool Function(SemanticsNode node);
 
+/// Signature for the [SemanticsConfiguration.merger].
+///
+/// The input map can be used for looking up [SemanticsConfiguration]s rendering
+/// children produces. The keys are [RenderObject]s, and the values are their
+/// [SemanticsConfiguration]s. The list of each value is always non-empty.
+///
+/// The return value is a [SemanticsMergeResult].
+typedef SemanticsConfigMerger = SemanticsMergeResult Function(Map<Object, List<SemanticsConfiguration>>);
```

Comment:
Do you always have to return all SemnaticsConfigurations from the map in the SemanticsMergeResult? What does it mean if you drop some of them on the floor? Is that a bug we should guard against?

Or should this really be just returning a `List<List<SemanticsConfiguration>>`, the siblings merge groups. And all SemnaticsConfigurations not included here are merged up as usual?

Reply:
The documented behavior would then be:

The SemanticsConfigMerger decides which SemanticsConfigurations of descendants should form explicit nodes. SemanticsConfigurations that are not made into explicit nodes are merged upwards and the explicit nodes are made into siblings of the upwards merge target.

Reply:
The fact that these explicit nodes are siblings and not children of the parent is really odd now that I see it written out explicitly. Are we sure we are doing the right thing here?

Reply:
> Do you always have to return all SemnaticsConfigurations from the map in the SemanticsMergeResult? What does it mean if you drop some of them on the floor? Is that a bug we should guard against?

I think dropping some configruations are ok. There will be use case where we want to insert additional synthetic configurations though.

If we return `List<List<SemanticsConfiguration>>` we will lose the functionality for both dropping or adding configurations.

> The fact that these explicit nodes are siblings and not children of the parent is really odd now that I see it written out explicitly.

This is for this specific use case, I don't have a use case where we want the merge group to be children of the parent. We can always add more api to SemanticsMergeRule if there is use case for it.